src/adapters/mson.es6

```diff
@@ -0,0 +1,201 @@
+/*
+ * Renders refract elements into MSON.
+ */
+
+/*
+ * Indent a piece of multiline text by a number of spaces.
+ * Setting `first` to `true` will also indent the first line.
+ */
+function indent(input, spaces, options={first: false}) {
+   let pre = '';
+   let lines = [];
+
+   for (let _ = 0; _ < spaces; _++) {
+     pre += ' ';
+   }
+
+   for (let line of input.split('\n')) {
+     // Only indent non-blank lines!
+     lines.push(line ? pre + line : line);
+   }
+
+   lines = lines.join('\n').trim();
+
+   if (options.first) {
+     lines = pre + lines;
+   }
+
+   return lines;
+ }
+
+/*
+ * Get type information for an element, such as the element name, whether
+ * it is required, etc. Returns an array of strings.
+ */
+function getTypeAttributes(element, attributes={}) {
+  let typeAttributes = [];
+
+  if (element.element !== 'string' && element.element !== 'dataStructure') {
+    typeAttributes.push(element.element);
+  }
+
+  return typeAttributes.concat(attributes.typeAttributes || []);
+}
+
+function handleContent(element, spaces, marker) {
+  let renderedContent = '';
+  let objectLike = null;
+
+  if (element.content === undefined) {
+    return [renderedContent, objectLike];
+  }
+
+  for (let item of element.content) {
+    // Note: the `initialIndent` option above works because we specifically
+    //       do *not* pass it down the rabbit hole here.
+    if (item.element === 'member') {
+      // This is an object type or something similar.
+      objectLike = true;
+      /* eslint-disable block-scoped-var */
+      renderedContent += handle(item.key.content, item.value, {
+        parent: element,
+        spaces,
+        marker,
+        attributesElement: item
+      });
+      /* eslint-enable block-scoped-var */
+    } else if (item.element === 'ref') {
+      renderedContent += `${marker} Include ${item.content.href}\n`;
+    } else if (item.element === 'select') {
+      // This is a `OneOf` mutually exclusive type. Currently not
+      // supported as it needs some support upstream in Minim.
+      console.warn('MSON select/option elements are not yet supported!');
+    } else {
+      // This is an array type or something similar.
+      objectLike = false;
+      /* eslint-disable block-scoped-var */
+      renderedContent += handle(item.meta.title.toValue(), item, {
+        parent: element,
+        spaces,
+        marker
+      });
+      /* eslint-enable block-scoped-var */
+    }
+  }
+
+  return [renderedContent, objectLike];
+}
+
+/*
+ * Handle the rendering of an element based on its element type. This function
+ * will call itself recursively to handle child elements for objects and
+ * arrays.
+ */
+function handle(name, element, {parent=null, spaces=4, marker='+',
```

Comment:
Note: I feel a functions over 100 lines of length are hard to read and should be broken down into multiple functions.